scania-digital-design-system/sdds
# feat(data table): event refactoring and removal of props

**Describe pull-request**  
This PR refactors the events in the data-table. It also removes the following props: 
`disableFilteringFunction`, `disableSortingFunction` and `enableClientPagination`

It introduces the following events:  `sddsPageChange`, `sddsSortChange`, `sddsFilterChange`
These can be prevented by a user to achieve the same effect as the props above. 

This is more inline with how other components work.

Also added a section in the migration docs to outline the breaking changes.

**Solving issue**  
Fixes: https://tegel.atlassian.net/browse/DTS-1479

**How to test**  
1. Go to storybook link below
2. Check in Data Table -> Web Components -> Sorting, Filtering, Pagination
3. Trigger the event for the corresponding story and look in to console for a log of the event. 
4. Check out the branch
5. Add preventDefault() to the event and see it fail. 

**Suggested test steps**
